slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunction.scala

```diff
@@ -17,24 +17,63 @@
 package za.co.absa.fadb.slick
 
 import slick.jdbc.{GetResult, SQLActionBuilder}
-import slick.jdbc.PostgresProfile.api._
-import za.co.absa.fadb.{DBFunctionFabric, QueryFunction}
+import za.co.absa.fadb.DBFunctionFabric
 
+/**
+  * Mix-in trait to use [[za.co.absa.fadb.DBFunction DBFunction]] with [[SlickPgEngine]]. Implements the abstract function `query`
+  * @tparam T - The input type of the function
+  * @tparam R - The return type of the function
+  */
 trait SlickPgFunction[T, R] extends DBFunctionFabric {
 
-  protected def sqlToCallFunction(values: T): SQLActionBuilder
+  /**
+    * A reference to the [[SlickPgEngine]] to use the [[za.co.absa.fadb.DBFunction DBFunction]] with
+    */
+  implicit val dbEngine: SlickPgEngine
 
-  protected def resultConverter: GetResult[R]
+  /**
+    * This is expected to return SQL part of the [[SlickQuery]] (eventually returned by the `SlickPgFunction.query` function
+    * @param values - the values to pass over to the database function
+    * @return       - the Slick representation of the SQL
+    */
+  protected def sql(values: T): SQLActionBuilder
 
-  protected def makeQueryFunction(sql: SQLActionBuilder)(implicit rconv: GetResult[R]): QueryFunction[Database, R] = {
-    val query = sql.as[R]
-    val resultFnc = {db: Database => db.run(query)}
-    resultFnc
+  /**
+    * This is expected to return a method to convert the [[slick.jdbc.PositionedResult slick.PositionedResult]], the Slick general SQL result
+    * format into the `R` type
+    * @return - the converting function
+    */
+  protected def slickConverter: GetResult[R]
+
+  /**
+    * Alias to use within the SQL query
+    */
+  protected val alias = "A"
```

Comment:
minor cosmetics, could be `FNC` or `F` or so, just to be more explicit about it - it's all about aliases to fields from function

Reply:
Haha, of course. It's just my first unspecific "go-to" alias